### PR TITLE
GEODE-6143: remove PowerMock from VMStatsMonitorTest

### PR DESCRIPTION
**Pull request by gesterzhou**
Thank you for submitting a contribution to Apache Geode.

@mcmellawatt 

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitorTest.java [hunk 1]
```diff
@@ -107,7 +89,7 @@ public void calculateCpuUsageShouldCorrectlyCalculateTheCpuUsed() {
   @Test
   public void refreshStatsShouldUpdateCpuUsage() {
     ZonedDateTime now = ZonedDateTime.now();
-    when(MBeanJMXAdapter.isAttributeAvailable(anyString(), anyString())).thenReturn(true);
+    vmStatsMonitor = new VMStatsMonitor(testName.getMethodName(), true);
```
Comment (mcmellawatt) [inline geode-core/src/test/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitorTest.java]: You are creating two VMStatsMonitor objects (line 92 and 93) where I think you meant to create one.  You can combine the two into just one line like:
```
vmStatsMonitor = spy(new VMStatsMonitor(testName.getMethodName(), true));
```

File: geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitor.java [hunk 2]
```diff
@@ -60,21 +60,24 @@ long getLastProcessCpuTime() {
...
     super(name);
-    processCPUTimeAvailable = MBeanJMXAdapter.isAttributeAvailable(PROCESS_CPU_TIME_ATTRIBUTE,
-        ManagementFactory.OPERATING_SYSTEM_MXBEAN_NAME);
+    this.processCPUTimeAvailable = processCPUTimeAvailable;
     if (!processCPUTimeAvailable) {
       cpuUsage = VALUE_NOT_AVAILABLE;
     }
   }
 
+  public VMStatsMonitor(String name) {
```
Comment (mcmellawatt) [inline geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitor.java]: Can we get rid of this constructor and just use the one above in the production code as well as the test code?  You could inject the `processCPUTimeAvailable` parameter in the production code as well using the above constructor.

File: geode-core/src/test/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitorTest.java (hunk 1 above)
Comment (gesterzhou) [reply geode-core/src/test/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitorTest.java]: fixed.

File: geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitor.java (hunk 2 above)
Comment (gesterzhou) [reply geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitor.java]: We cannot. The callers are in different packages.

**Pull request merged**